kytos/core/switch.py

```diff
@@ -207,6 +226,23 @@ def __init__(self, address, port, socket, switch=None):
         self.port = port
         self.socket = socket
         self.switch = switch
+        self._state = None
+        self.STATES = Connection_states
```

Comment:
If we are going to use the setter here, the best approach would be something like this:
```
class Test():
    def __init__(self):
        self.x = None
    @property
    def x(self):
        return self.__x
    @x.setter
    def x(self, x):
        self.__x = x
        print('setting x with {}'.format(x))
```